feat(pie-docs): DSW-2496 radio-group overview page (#2053)
justeattakeaway/pie: apps/pie-docs/src/components/radio-group/overview/overview.md

---
eleventyNavigation:
    key: Overview
    parent: 'Radio Group'
    order: 1
shouldShowContents: true
permalink: components/radio-group/
---

## Overview

> Radio groups offer users a single selection method,  Each radio button functions independently, providing immediate feedback through its selected or unselected state.
> 
> These groups are frequently utilised in forms, settings panels, and various interfaces requiring users to make a single selection or provide input from predefined options.

Reviewer: ```suggestion
Radio groups present a set of mutually exclusive options, allowing users to select only one option from a predefined list. They are commonly used in forms, settings panels, and other interfaces where users need to make a single selection from multiple options.
```

{% contentPageImage {
    src:"../../../assets/img/components/radio-group/overview.svg",
    alt: "A group of radio buttons that are vertically stacked together."
} %}

---

## Dos and Don’ts

{% usage {
    do: {
        type: usageTypes.text,
        items: [

>             "Ensure each radio button has a concise and descriptive label that clearly indicates what selecting it will entail.",

Reviewer: ```suggestion
            "Make sure each radio button has a label which is clear, concise and accurate.",
```

            "Use a minimum of 2 radio buttons within a group."
        ]
    },
    dont: {
        type: usageTypes.text,
        items: [

>             "If the user can select multiple option from a group, checkboxes should be used instead of radio buttons.",

Reviewer: ```suggestion
            "If the user can select multiple options from a group, checkboxes should be used instead of radio buttons.",
```

            "Limit the number of radio buttons presented to users to avoid overwhelming them with choices."
        ]
    }
} %}

---

## Anatomy

{% contentPageImage {
    src:"../../../assets/img/components/radio-group/anatomy.svg",
    alt: "Anatomy of a radio group.",
    width: 340
} %}

{% list {
    type: listTypes.ordered,
    items: [

>         "**Form label (Optional):** The [form label](/components/form-label/) provides necessary content / information to a form.",

Reviewer: Thoughts?
```suggestion
        "**Form label (Optional):** The [form label](/components/form-label/) provides a clear and accessible description of the form element it is associated with.",
```

>         "**Radio input:** A radio input indicates the appropriate state. By default it is unselected.",
>         "**Radio label:** Describes the information you want to select or unselect.",
>         "**Assistive text (Optional):** The [assistive text](/components/assistive-text/) provides additional instructional information / error / success messaging for the whole radio group."

Reviewer: ```suggestion
        "**Radio input:** A radio input indicates the current selected state for the option. By default it is unselected.",
        "**Radio label:** Describes the option that can be selected.",
        "**Assistive text (Optional):** The [assistive text](/components/assistive-text/) provides additional instructions and error or success messaging for the entire radio group."
```

    ]
} %}

---

## Variations
Radio groups can be laid out both horizontally and vertically depending on the use cas and the structure of the UI. Where possible, arrange the radio group vertically for easier reading.

### Horizontal
{% contentPageImage {
    src:"../../../assets/img/components/radio-group/variations-horizontal.svg",
    alt: "A row of radio buttons grouped together by a single label above them.",
    width: 258
} %}

### Vertical
{% contentPageImage {
    src:"../../../assets/img/components/radio-group/variations-vertical.svg",
    alt: "A vertical stack of radio buttons grouped together by a single label above them.",
    width: 257
} %}

---

## Modifiers

### Form label

> A form label is recommended to provide the user context of the form element and it’s content.

Reviewer: ```suggestion
A form label is recommended to provide the user context of the form element and its content.
```

    {% contentPageImage {
      src:"../../../assets/img/components/radio-group/modifiers-form-label-2.svg",
      alt: "A row of radio buttons grouped together but missing a label.",
      width: 258
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

### Assistive text
Assistive text is used where needed to provide additional information or error / success messaging regarding the whole form element, and never to individual checkboxes. The assistive text is always used for an error state to provide the user with the information required to complete the form element.

{% contentPageImage {
  src:"../../../assets/img/components/radio-group/modifiers-assistive-text.svg",
  alt: "A row of radio buttons grouped with an error colour around their borders. Below is a message with an error icon. The colour used for both the text and icon is the same as the error border.",
  width: 258
} %}

---

## Content
All content should use sentence case.

### Form label
{% notification {
  type: "neutral",
  iconName: "link",
  message: "Check out the [form label content](/components/form-label/) documentation."
} %}

### Label
- Always use clear and concise labels for checkboxes.

> - Labels appear to the right of checkbox input.

Reviewer: Radios?

{% contentLayout %}
  {% contentItem %}
    {% contentPageImage {
      src:"../../../assets/img/components/radio-group/states-1.svg",
      alt: "A row of radio buttons grouped with an error colour around their borders. Below is a message with an error icon. The colour used for both the text and icon is the same as the error border.",
      width: 258
    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src:"../../../assets/img/components/radio-group/states-2.svg",
      alt: "A row of radio buttons grouped together by a single label above them. Their background colours are faded to signify that they are currently in a disabled state.",
      width: 258
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

---

## Examples

### LTR examples
Here are some examples of the component in left-to-right context:

{% contentPageImage {
  src:"../../../assets/img/components/radio-group/ltr.svg",
  alt: "A vertical stack of radio buttons. The first one is selected. The radio buttons sit on the left hand side of the label text. This demonstrates a left-to-right reading direction.",
  width: 257
} %}

### RTL examples

> Here are some examples of the component in right-to-left context:

Reviewer: ```suggestion
Here are some examples of the component in a right-to-left context:
```
